beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -115,6 +104,48 @@ func (s *Service) validateCommitteeIndexBeaconAttestation(ctx context.Context, p
 		}
 	}
 
+	if !s.cfg.chain.InForkchoice(bytesutil.ToBytes32(data.BeaconBlockRoot)) {
```

Comment:
Executing all these methods before we have validated `data.BeaconBlockRoot` is wrong. As we might have not processed the block yet (and it has to be inserted into our pending queue)